src/components/AppConfig/AppConfig.tsx

```diff
@@ -20,9 +20,35 @@ export interface LLMGatewaySettings {
   url?: string;
 }
 //////////////////////
+
+export async function saveLLMOptInState(optIn: boolean): Promise<boolean> {
+  return lastValueFrom(
+    getBackendSrv().fetch({
+      url: `api/plugins/grafana-llm-app/resources/save-llm-state`,
+      method: 'POST',
+      data: { optIn },
+    })
+  )
+    .then((response: FetchResponse) => {
+      if (!response.ok) {
+        console.error(`Error using Grafana-managed LLM: ${response.status} ${response.data.message}`);
```

Comment:
One thing to consider, would be to show these failed results to the user via an alert. Not a blocker though.

Reply:
Afaik these should pop up as an toast notification depending on the response: warning for 4xx, error for 5xx responses (at least on my end)

<img width="441" alt="Screenshot 2024-02-07 at 15 58 47" src="https://github.com/grafana/grafana-llm-app/assets/13647408/5fecacae-9532-4791-b330-8d2307c3e2b9">